src/js/index.js

```diff
@@ -51,3 +53,81 @@ if ('serviceWorker' in navigator) {
       didUpdate();
     });
 }
+
+search.initialize(function (err, index) {
```

Comment:
A promise is sort-of hard to use internally, but the external interface can definitely be a promise.